### ISSUE TITLE
Support Neorg

### ISSUE THREAD

**Issue by austinbutler** [ltex-plus/ltex-ls-plus#55]
Was previously requested on the original repo: https://github.com/valentjn/ltex-ls/issues/258

It's somewhat similar to the already-supported Org.

**Comment by spitzerd** [comment]
Done. You can start testing with the next nightly release.

**Comment by austinbutler** [comment]
Thank you!

Got it working, but from looking at the commit I thought I'd just enable `neorg`. That didn't work, had to also enable `norg`.

I use [Neovim with lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#ltex_plus), and these are my settings right now:

```lua
local lspconfig = require("lspconfig")
lspconfig.ltex_plus.setup({
	capabilities = cmp_capabilities,
	filetypes = {
		"bib",
		"context",
		"html",
		"markdown",
		"neorg",
		"norg",
		"org",
		"pandoc",
		"plaintex",
		"quarto",
		"mail",
		"mdx",
		"rmd",
		"rnoweb",
		"rst",
		"tex",
		"text",
		"typst",
		"xhtml",
	},
	settings = {
		ltex = {
			enabled = {
				"bib",
				"context",
				"html",
				"markdown",
				"neorg",
				"norg",
				"org",
				"pandoc",
				"plaintex",
				"quarto",
				"mail",
				"mdx",
				"rmd",
				"rnoweb",
				"rst",
				"tex",
				"text",
				"typst",
				"xhtml",
			},
		},
	},
})
```

With that it kicks in. Problems I've noticed so far:

1. It checks for spelling mistakes (`MORFOLOGIK_RULE_EN_US`) and `Two consecutive dots`  (`DOUBLE_PUNCTUATION`) inside of URLs
   - So `{https://github.com/ltex-plus/ltex-ls-plus/pulls?q=is%3Apr+is%3Aopen+merged%3A2024-01-01..2025-12-31}[Pull requests]` would complain about the two dots in the date range
   - `{:marketing-plans:}[Marketing Plans]` also complains about the spelling of `marketing-plans`
1. It checks spelling inside backticks and code blocks
   - For example, ``Here's a `spellign` mistake``, Markdown would ignore it since it's in backticks but Neorg checks within the backticks
   - Example code block:
    ```
    @code
    print("Hrlo worls")
    @end
    ```
1. Using Neorg's incomplete task syntax triggers `Don't put a space after the opening parenthesis` (`COMMA_PARENTHESIS_WHITESPACE`)
   - `( ) Update task board`
1. Indented text under a heading results in `you repeated a whitespace` (`.WHITESPACE_RULE`)
   - Basically if you have a heading (`* My heading`), anything under it will be indented with two spaces to visually show it under the heading (so it will be like `  Text under header`)
